### PR TITLE
Merge Api Basics to Main

### PR DESCRIPTION
**Pull request by Xen0Xys**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+PORT=3000
+BIND_ADDRESS=0.0.0.0
+
+# SSL settings
+SSL=false
+SSL_KEY_FILE=server.key
+SSL_CERT_FILE=server.crt
+
+# Database settings
+DB_TYPE=sqlite
```
Comment (loicmaes) [inline .env.example]: Maybe add more documentation (valid/accepted database types?)

File: src/app.js [hunk 2]
```diff
@@ -1 +1,46 @@
...
+    standardHeaders: true,
+    legacyHeaders: false,
+}));
+
+// TODO: Route registering
+// require("./handlers/RoutesHandler")(app);
+
+// Default route
+app.get("/", (req, res) => {
+    res.status(404).json({error: "Not found"});
```
Comment (loicmaes) [inline src/app.js]: Why the default route is responding with an error?

File: src/app.js [hunk 3]
```diff
@@ -1 +1,46 @@
...
+});
+
+// Start server
+if(process.env.SSL.toUpperCase() === "FALSE"){
```
Comment (loicmaes) [inline src/app.js]: I'm not sure that's necessary; maybe try only
```js
if (process.env.SSL) {
    ...
} else {
    ...
}
```

File: src/app.js [hunk 4]
```diff
@@ -1 +1,46 @@
...
+    app.listen(parseInt(process.env.PORT), process.env.BIND_ADDRESS, () => {
+        console.log(`Server started on http://${process.env.BIND_ADDRESS}:${process.env.PORT}`);
```
Comment (loicmaes) [inline src/app.js]: Add a global log function?

File: src/app.js [hunk 5]
```diff
@@ -1 +1,46 @@
...
+    });
+}else{
+    https.createServer({
+        key: fs.readFileSync(process.env.KEY_FILE),
+        cert: fs.readFileSync(process.env.CERT_FILE)
+    }, app).listen(parseInt(process.env.PORT), () => {
+        console.log(`Server started on https://${process.env.BIND_ADDRESS}:${process.env.PORT}`);
```
Comment (loicmaes) [inline src/app.js]: Use the global log function to log easily

File: src/database/sequelize.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        dialect: process.env.DB_TYPE,
+        logging: process.env.DB_LOGGING === "true"
+    });
+
+
+// Loading models
+// const models = require('../handlers/modelHandler')(sequelize);
+const models = undefined; // TODO: Load models
+
+sequelize.sync({force: process.env.DB_FORCE_INIT === "true"}).then(() => {
```
Comment (loicmaes) [inline src/database/sequelize.js]: Same as before, maybe try:
```js
if (process.env.DB_FORCE_INIT) {
    ...
}
```

File: src/database/sequelize.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        dialect: "sqlite",
+        storage: "database.db",
+        logging: process.env.DB_LOGGING === "true"
+    });
+else
+    sequelize = new Sequelize(process.env.DB_NAME, process.env.DB_USER, process.env.DB_PASSWORD, {
+        host: process.env.DB_HOST,
+        port: process.env.DB_PORT,
+        dialect: process.env.DB_TYPE,
+        logging: process.env.DB_LOGGING === "true"
```
Comment (loicmaes) [inline src/database/sequelize.js]: Use JS string booleans to manage it easily (no case troubles)

File: src/database/sequelize.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-undef */
+const { Sequelize } = require("sequelize");
+
+let sequelize;
+if(process.env.DB_TYPE === "sqlite")
+    sequelize = new Sequelize({
+        dialect: "sqlite",
+        storage: "database.db",
+        logging: process.env.DB_LOGGING === "true"
```
Comment (loicmaes) [inline src/database/sequelize.js]: Use JS string booleans to manage it easily (no case troubles)

File: src/database/sequelize.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-undef */
+const { Sequelize } = require("sequelize");
+
+let sequelize;
+if(process.env.DB_TYPE === "sqlite")
```
Comment (loicmaes) [inline src/database/sequelize.js]: Maybe fix case issues? (add a toLower or a toUpper)

File: src/app.js (hunk 4 above)
Comment (Xen0Xys) [reply src/app.js]: What did you mean by global log function ? Do you have an example ?

File: src/app.js (hunk 3 above)
Comment (Xen0Xys) [reply src/app.js]: After testing, `if (process.env.SSL)` is `true` if there is a string, no matter the string value.

File: src/app.js (hunk 4 above)
Comment (loicmaes) [reply src/app.js]: Yes, you can simply use an arrow function to log some things:
```js
// 'secure' is a boolean (based on your SSL system)
const log = (secure) => console.log(`${secured ? 'https' : 'http'}://...`)
```

File: package.json [hunk 10]
```diff
@@ -27,10 +27,16 @@
   "homepage": "https://github.com/AmicaleCORE/Karl-API#readme",
   "devDependencies": {
     "eslint": "^8.36.0",
-    "nodemon": "^2.0.22"
+    "nodemon": "^2.0.22",
+    "sqlite3": "^5.1.6"
```
Comment (loicmaes) [inline package.json:31]: You need to install mariadb and mysql as well.
```
npm i --save mysql2 mariadb
```
These modules are supported by Sequelize and help you to interact with few more databases.

File: package.json [hunk 11]
```diff
@@ -27,10 +27,20 @@
...
   "dependencies": {
+    "cors": "^2.8.5",
+    "dotenv": "^16.0.3",
     "express": "^4.18.2",
+    "express-rate-limit": "^6.7.0",
+    "helmet": "^6.0.1",
+    "https": "^1.0.0",
+    "mariadb": "^3.1.1",
+    "mysql2": "^3.2.0",
+    "pg": "^8.10.0",
```
Comment (loicmaes) [inline package.json:42]: What's the 'pg' module?

**Review by loicmaes — APPROVED** [review]

File: package.json (hunk 11 above)
Comment (Xen0Xys) [reply package.json:42]: `pg` is for Postgres database.

**Pull request merged**